plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestMetadataQueryOptimization.java

```diff
@@ -118,6 +119,30 @@ public void testOptimization()
                         values(ImmutableList.of("b", "c"), ImmutableList.of())));
     }
 
+    @Test
+    public void testOptimizationWithNullPartitions()
+    {
+        String testTable = "test_metadata_optimization_with_null_partitions";
+
+        getPlanTester().executeStatement(format(
+                "CREATE TABLE %s (a, b, c) WITH (PARTITIONING = ARRAY['b', 'c'])" +
+                        "AS VALUES (5, 6, CAST(NULL AS INTEGER)), (8, 9, CAST(NULL AS INTEGER))",
+                testTable));
+
+        Session session = Session.builder(getPlanTester().getDefaultSession())
+                .setSystemProperty("optimize_metadata_queries", "true")
```

Comment:
@raunaqmorarka we've been speaking already on `optimize_metadata_queries` session property for Iceberg/ Delta Lake offline.
We should figure out a way where it is not necessary for the user to specify explicitly this low level session property to make use of https://trino.io/docs/current/admin/properties-optimizer.html#optimizer-optimize-metadata-queries